src/rules/class_literal_property_style.rs

```diff
@@ -0,0 +1,762 @@
+
+    use super::*;
+
+    #[test]
+    fn test_class_literal_property_style_rule() {
+        RuleTester::run(
+            class_literal_property_style_rule(),
+            rule_tests! {
+                valid => [
+                  r#"
+              class Mx {
+                declare readonly p1 = 1;
+              }
+                  "#,
+                  r#"
+              class Mx {
+                readonly p1 = 'hello world';
+              }
+                  "#,
+                  r#"
+              class Mx {
+                p1 = 'hello world';
+              }
+                  "#,
+                  r#"
+              class Mx {
+                static p1 = 'hello world';
+              }
+                  "#,
+                  r#"
+              class Mx {
+                p1: string;
+              }
+                  "#,
+                  r#"
+              class Mx {
+                get p1();
+              }
+                  "#,
+                  r#"
+              class Mx {
+                get p1() {}
+              }
+                  "#,
+                  r#"
+              abstract class Mx {
+                abstract get p1(): string;
+              }
+                  "#,
+                  r#"
+                    class Mx {
+                      get mySetting() {
+                        if (this._aValue) {
+                          return 'on';
+                        }
+
+                        return 'off';
+                      }
+                    }
+                  "#,
+                  r#"
+                    class Mx {
+                      get mySetting() {
+                        return `build-${process.env.build}`;
+                      }
+                    }
+                  "#,
+                  r#"
+                    class Mx {
+                      getMySetting() {
+                        if (this._aValue) {
+                          return 'on';
+                        }
+
+                        return 'off';
+                      }
+                    }
+                  "#,
+                  r#"
+                    class Mx {
+                      public readonly myButton = styled.button`
+                        color: ${props => (props.primary ? 'hotpink' : 'turquoise')};
+                      `;
+                    }
+                  "#,
+                  {
+                    code => r#"
+                      class Mx {
+                        public get myButton() {
+                          return styled.button`
+                            color: ${props => (props.primary ? 'hotpink' : 'turquoise')};
+                          `;
+                        }
+                      }
+                    "#,
+                    options => "fields",
+                  },
+                  // TODO: uncomment if
+                  // https://github.com/tree-sitter/tree-sitter-typescript/issues/270 gets
```

Comment:
It looks like the tree-sitter-typescript grammar doesn't parse eg `public declare`, so opened https://github.com/tree-sitter/tree-sitter-typescript/issues/270